src/jit/codegenxarch.cpp

```diff
@@ -188,7 +224,13 @@ void                CodeGen::genEmitGSCookieCheck(bool pushReg)
         if (compiler->lvaKeepAliveAndReportThis() && compiler->lvaTable[compiler->info.compThisArg].lvRegister &&
             (compiler->lvaTable[compiler->info.compThisArg].lvRegNum == REG_ECX))
         {
```

Comment:
Even this check needs modification since thisPtr the first arg on Linux will be in REG_EDI.  Here we should replace REG_ECX with REG_ARG_0